File: LeetCode/TwoSum/CPP/TwoSum.cpp
//
// Created by Shahriar Nasim Nafi on 18/10/21.
//

#include <iostream>

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> fnums;
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    if (i != j) {
                        fnums.push_back(i);
                        fnums.push_back(j);
                        return fnums;
                    }
                }
            }
        }
        return fnums;
    }
};

File: LeetCode/TwoSum/Swift/TwoSum.swift
//
// Created by Shahriar Nasim Nafi on 30/12/21.
//
// Problem: https://leetcode.com/problems/two-sum/
//

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        for i in 0..<(nums.count - 1) {
            for j in stride(from: 0, to: (nums.count), by: 1) {
                 if nums[i] + nums[j] == target {
                     if (i != j) {
                         return [i, j]
                    }
                 }
            }
        }
        return []
    }
}